neo/SmartContract/InteropService.cs

```diff
@@ -143,8 +143,41 @@ private static bool Runtime_GetTrigger(ApplicationEngine engine)
 
         internal static bool CheckWitness(ApplicationEngine engine, UInt160 hash)
         {
-            var _hashes_for_verifying = engine.ScriptContainer.GetScriptHashesForVerifying(engine.Snapshot);
-            return _hashes_for_verifying.Contains(hash);
+            if (engine.ScriptContainer is Transaction tx)
+            {
+                foreach (var attribute in tx.Attributes)
+                {
+                    if (attribute.Usage != TransactionAttributeUsage.Cosigner) continue;
+                    CosignerUsage usage = attribute.Data.AsSerializable<CosignerUsage>();
+                    if (usage.ScriptHash != hash) continue;
+
+                    switch (usage.Scope.Type)
+                    {
+                        case WitnessScopeType.Global: return true;
```

Comment:
This is problematic... 
imagine, we add one attribute which is global, and it ends up "authorizing" another attached witness, which is not global.

Reply:
Check my last PR

Reply:
What PR?